merkle/log_proofs.go

```diff
@@ -62,8 +61,7 @@ func CalcInclusionProofNodeAddresses(snapshot, index, treeSize int64) ([]NodeFet
 	if index < 0 {
 		return nil, status.Errorf(codes.InvalidArgument, "invalid parameter for inclusion proof: index %d is < 0", index)
 	}
-
-	return pathFromNodeToRootAtSnapshot(index, 0, snapshot, treeSize)
+	return proofNodes(uint64(index), 0, uint64(snapshot), snapshot < treeSize), nil
```

Comment:
Could you add a comment explaining why `recompute = snapshot < treeSize` ?

Reply:
Done. Btw, this `rehash` parameter is only for compatibility with the previous implementation. It can be set to `true` unconditionally, and everything would still work. This is how things will work in storage with not revisions.